tests/core/engine_adapter/test_trino.py

```diff
@@ -148,26 +151,36 @@ def test_partitioned_by_iceberg_transforms(
         return_value="datalake_iceberg",
     )
 
-    columns_to_types = {
-        "cola": exp.DataType.build("INT"),
-        "colb": exp.DataType.build("TEXT"),
-    }
+    expressions = d.parse(
+        f"""
+        MODEL (
+            name test_table,
+            partitioned_by (day(cola), truncate(colb, 8), colc),
+            kind INCREMENTAL_BY_TIME_RANGE(
+                time_column cola,
+            ),
+        );
+
+        SELECT 1::timestamp AS cola, 2::varchar as colb, 'foo' as colc;
+    """
+    )
+    model: SqlModel = t.cast(SqlModel, load_sql_based_model(expressions))
 
     adapter.create_table(
-        "test_table",
-        columns_to_types,
-        partitioned_by=[exp.to_column("day(cola)"), exp.to_column("truncate(colb, 8)")],
```

Comment:
The existing test "cheated" by doctoring values for `partitioned_by` that dont actually reflect what the parser would produce when parsing a SQL Model.

So I reimplemented it based on specifying a model string more like a user would